cmd/finch/nerdctl.go

```diff
@@ -121,10 +126,27 @@ func (nc *nerdctlCommand) run(cmdName string, args []string) error {
 				arg = fmt.Sprintf("%s%s", arg[0:11], resolvedIP)
 			}
 			nerdctlArgs = append(nerdctlArgs, arg)
+		case strings.HasPrefix(arg, "-v") || strings.HasPrefix(arg, "--volume"):
+			if dockerCompat {
+				volume, shouldSkip := getVolume(arg, args[i+1])
+				volumes = append(volumes, volume)
+				skip = shouldSkip
+			} else {
+				nerdctlArgs = append(nerdctlArgs, arg)
+			}
 		default:
 			nerdctlArgs = append(nerdctlArgs, arg)
 		}
 	}
+
+	if dockerCompat {
+		// Reverse sort to append to beginning.
+		volumes = reverseSortVolumes(volumes)
+		for _, vol := range volumes {
+			nerdctlArgs = append([]string{"-v", vol}, nerdctlArgs...)
```

Comment:
Looks like some tests may be failing due to using the shorthand flag?

>  time="2023-05-26T15:17:19Z" level=fatal msg="unknown shorthand flag: 'v' in -v"

[Source](https://github.com/runfinch/finch/actions/runs/5092355040/jobs/9153592038?pr=417#step:7:1672)

Reply:
This happens in all the subcommand use case :(
Unforturnately the subcommands are included in args.
```
func (nc *nerdctlCommand) run(cmdName string, args []string) error {
	nc.logger.Info(cmdName)
```

```
➜  finch git:(sort) ✗ ./_output/bin/finch container run
INFO[0000] container                                    
FATA[0000] requires at least 1 arg(s), only received 0  
FATA[0001] exit status 1                                
➜  finch git:(sort) ✗ ./_output/bin/finch run          
INFO[0000] run                                          
FATA[0000] requires at least 1 arg(s), only received 0  
FATA[0000] exit status 1   
```

Reply:
Will insert to the first index of -v